### PR TITLE
user settings

### PR DESCRIPTION
**Pull request by GusCaplan**

### CODE REVIEW

**Comment by iCrawl** [comment]
Note: Currently ClientUser.settings already exists, but only in master. Since we haven't released master yet we can safely overwrite the functionality of it to a better one than just the RAW output from discord in an object.

File: src/structures/ClientUserSettings.js [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+   * @param {number} position Absolute or relative position
+   * @param {boolean} [relative=false] Whether to position relatively or absolutely
+   * @returns {Promise<Guild>}
+   */
+  setGuildPosition(guild, position, relative) {
+    const temp = Object.assign([], this.guildPositions);
+    Util.moveElementInArray(temp, guild.id, position, relative);
+    return this.update('guild_positions', temp).then(() => guild);
+  }
+
```
Comment (PgBiel) [inline src/structures/ClientUserSettings.js]: here

File: src/structures/ClientUserSettings.js [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+  addRestrictedGuild(guild) {
+    const temp = Object.assign([], this.restrictedGuilds);
+    if (temp.includes(guild.id)) return Promise.reject(new Error('Guild is already restricted'));
+    temp.push(guild.id);
+    return this.update('restricted_guilds', temp).then(() => guild);
+  }
+
```
Comment (PgBiel) [inline src/structures/ClientUserSettings.js]: and here

File: src/util/Constants.js [hunk 3]
```diff
@@ -372,6 +376,142 @@ exports.DefaultAvatars = {
...
+     * @name ClientUserSettings#explicitContentFilter
+     * @type {string}
+     */
+    return exports.ExplicitContentFilterTypes[type];
+  },
+  friend_source_flags: function friendSources(flags) { // eslint-disable-line func-name-matching
+    /**
+     * Who can add you as a friend
+     * @name ClientUserSettings#friendSources
+     * @type {Object}
```
Comment (LanaCoyote) [inline src/util/Constants.js]: Can you add the properties for this object to the documentation?

File: src/client/websocket/packets/handlers/UserSettingsUpdate.js [hunk 4]
```diff
@@ -0,0 +1,18 @@
+const AbstractHandler = require('./AbstractHandler');
+const Constants = require('../../../../util/Constants');
+
+class UserSettingsUpdateHandler extends AbstractHandler {
```
Comment (iCrawl) [inline src/client/websocket/packets/handlers/UserSettingsUpdate.js]: Docs?

File: src/structures/ClientUserSettings.js [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+/**
+ * A wrapper around the ClientUser's settings
+ */
+class ClientUserSettings {
+  constructor(user, data) {
+    this.user = user;
+    this.patch(data);
+  }
+
+  patch(data) {
```
Comment (iCrawl) [inline src/structures/ClientUserSettings.js]: Docs?

File: src/client/websocket/packets/handlers/UserSettingsUpdate.js (hunk 4 above)
Comment (GusCaplan) [inline src/client/websocket/packets/handlers/UserSettingsUpdate.js]: none of the other handlers are documented

**Pull request closed without merging**